### PR TITLE
Ignore composes declaration for color-named

### LINKED ISSUE #1762: Add ignoreProperties: [] option to color-named
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

When you add a color name (which is a CSS class) in `composes`, it triggers a warning.

> Which rule, if any, is this issue related to?

`color-named`

> What CSS is needed to reproduce this issue?

```css
.Grid {
  composes: relative blue from "../index.css";
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "color-named": "never"
  }
}
```

> Which version of stylelint are you using?

`7.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

"CLI with `./node_modules/stylelint/dist/cli.js ./src/**/*.css`"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

"Yes, it's related to CSS modules."

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors you are getting)?

"The following warnings were flagged:"

```shell
src/Grid/Grid.css
Unexpected named color "blue"
```

Relates to #1724 #1712
Ideally, we would be able to have a similar fix by adding `composes` to the following list:
https://github.com/stylelint/stylelint/blob/9807f6de97625bacd8f9b3b74b300cb9eb1b459a/src/reference/propertySets.js

### PR DESCRIPTION
**Pull request by christianvuerings**
Fixes #1762

### CODE REVIEW

**Comment by jeddy3** [comment]
@christianvuerings Thanks for the detailed report and for the PR.

As `composes` isn't a standard property I'm not sure about adding it to our *custom idents* reference set. What we normally do in these situations is add an `ignore*` secondary option, as that allows users to ignore future non-standard properties too.

Here's [a recent pull request](https://github.com/stylelint/stylelint/pull/1534) that does that for another rule. The tests can be simplified to something like [this](https://github.com/stylelint/stylelint/blob/master/src/rules/property-no-unknown/__tests__/index.js#L88-L113) and the docs to something like [this](https://github.com/stylelint/stylelint/tree/master/src/rules/property-no-unknown#optional-options). 

Please can you update your PR?

**Comment by jeddy3** [comment]
@chrislloyd and @akleiner2 is this something either of you two would like to contribute?

**Comment by christianvuerings** [comment]
@jeddy3 pushed up an update, could you have another look at this?

**Comment by christianvuerings** [comment]
@jeddy3 build should be green now, after using `optionsMatches`

File: src/rules/color-named/README.md [hunk 1]
```diff
@@ -105,3 +105,21 @@ a { color: $blue; }
 ```less
 a { color: @blue; }
 ```
+
+## Optional options
+
+### `ignoreProperties: ["/regex/", "string"]`
+
+Given:
```
Comment (jeddy3) [inline src/rules/color-named/README.md]: Can you please change this to:

----

For example, with `"never"`.

Given:

```js
["/^my-/", "composes"]
```

The following patterns are *not* considered warnings:

```css
a {
  my-property: red;
}
```

```css
a {
  my-other-property: red;
}
```

```css
a {
  composes: red from './index.css';
}
```

---

File: src/rules/color-named/__tests__/index.js [hunk 2]
```diff
@@ -258,3 +258,53 @@ testRule(rule, {
     column: 23,
   } ],
 })
+
+testRule(rule, {
```
Comment (jeddy3) [inline src/rules/color-named/__tests__/index.js]: Can you please change your tests to just:

```js
testRule(rule, {
  ruleName,
  config: [ "never", { ignoreProperties: ["composes", "/^my-/"] } ],
  skipBasicChecks: true,

  accept: [ {
    code: "a { composes: blue from 'src/index.css'; }",
  }, {
    code: "a { composes: grey blue from 'src/index.css'; }",
  }, {
    code: "a { my-property: red; }",
  }, {
    code: "a { my-other-property: blue; }",
  } ],

  reject: [ {
    code: "a { color: rebeccapurple; }",
    message: messages.rejected("rebeccapurple"),
    line: 1,
    column: 12,
  }, {
    code: "a { not-my-property: red; }",
    message: messages.rejected("red"),
    line: 1,
    column: 21,
  } ],
})
```

File: src/rules/color-named/__tests__/index.js [hunk 3]
```diff
@@ -258,3 +258,53 @@ testRule(rule, {
...
+    column: 12,
+  }, {
+    code: "a { background: #00c, red, #fff; }",
+    message: messages.rejected("red"),
+    line: 1,
+    column: 23,
+  } ],
+})
+
+testRule(rule, {
```
Comment (jeddy3) [inline src/rules/color-named/__tests__/index.js]: You can remove these tests as they're a bit redundant.

**Comment by jeddy3** [comment]
@christianvuerings Thank you for adding this!

The rule code looks great.

I've requested changes to the docs and tests though. There are a lot of tests and rule READMEs in stylelint, as such we're making an effort to add as much consistency to them. To the READMEs for the benefit of the user, and the tests for the benefit of the maintainers.

**Comment by christianvuerings** [comment]
@jeddy3 Made the requested updates and made sure all of the tests pass (had to update 1 suggested `column` property to `22` instead of `21`)

One of the (many) reasons why I like Stylelint is because of the awesome doc & test coverage, so definitely willing to do my part!

**Comment by jeddy3** [comment]
> Made the requested updates 

Thanks you.

> I like Stylelint is because the awesome doc & test coverage, so definitely willing to do my part!

That's fantastic on both accounts.

One last request, can you add please a changelog item e.g. "Added: `ignoreProperties` option to `color-named`"?

@stylelint/core Once the changelog item is done, anyone see any issues with merging this in?

**Comment by m-allanson** [comment]
@christianvuerings thanks for adding this :+1: 

The changes look good to me.

**Comment by christianvuerings** [comment]
@jeddy3 & @m-allanson also updated the changelog

**Comment by jeddy3** [comment]
LGTM

**Comment by christianvuerings** [comment]
@jeddy3 & @m-allanson are there any other changes required before merging this PR?

**Comment by m-allanson** [comment]
@christianvuerings I'm not sure if there's an official policy on this, but generally any non-trivial PR will get approval from @jeddy3 and @davidtheclark (the two stylelint owners) before being merged in.

**Comment by christianvuerings** [comment]
@m-allanson thanks for explaining. I'll wait for the approval from @davidtheclark

File: src/rules/color-named/README.md [hunk 4]
```diff
@@ -105,3 +105,35 @@ a { color: $blue; }
 ```less
 a { color: @blue; }
 ```
+
+## Optional options
```
Comment (ntwb) [inline src/rules/color-named/README.md]: Can this be updated to `Optional secondary options` please.

This was standardised in #1816

**Comment by davidtheclark** [comment]
LGTM. After the one docs wording change is taken care of we can merge, I think.

**Comment by christianvuerings** [comment]
@m-allanson @jeddy3 @davidtheclark just pushed the update for the `Optional secondary options` change.

**Comment by jeddy3** [comment]
@christianvuerings Thanks for this!

**Pull request merged**